pkg/arvo/app/publish.hoon

```diff
@@ -285,10 +285,41 @@
     ^-  (quip card _this)
     ?-    -.sin
         %poke-ack   (on-agent:def wir sin)
+    ::  If our subscribe failed, delete notebook associated with subscription if
+    ::  it exists
     ::
-        %watch-ack  (on-agent:def wir sin)
+        %watch-ack
+      ?.  ?=([%subscribe @ @ ~] wir)
+        (on-agent:def wir sin)
+      ?~  p.sin
+        [~ this]
+      =/  who=@p     (slav %p i.t.wir)
+      =/  book=@tas  i.t.t.wir
+      [~ this(subs (~(del by subs) who book))]
+    ::  Resubscribe to any subscription we get kicked from. The case of actually
+    ::  getting banned from a notebook is handled by %watch-ack
     ::
-        %kick       (on-agent:def wir sin)
+        %kick
```

Comment:
If you aren't ever modifying this, why not break it out of the statement with :_  this right before line 303?